src/App.vue

```diff
@@ -95,35 +107,73 @@ export default defineComponent({
         },
         completed(): Todo[] {
             return this.todos.filter(todo => todo.isCompleted);
+        },
+        isAppReady(): boolean {
+            return !this.isLoadingUserSession && !this.isLoadingTodos;
+        },
+        isLoggedIn(): boolean {
+            return Boolean(this.user.uid);
+        },
+        userTodosPath(): string {
+            return `users/${this.user.uid}/todos`;
         }
     },
+    created() {
+        this.setUser();
+    },
     methods: {
+        setUser() {
+            firebaseApp.auth().onAuthStateChanged((user: User) => {
+                this.isLoadingUserSession = false;
+
+                if (user) {
+                    this.user = user;
+                    this.getUserTodos();
+                } else {
+                    this.isLoadingTodos = false;
+                }
+            })
+        },
+        getUserTodos() {
+            db.ref(this.userTodosPath).on("value", (snap: any) => {
```

Comment:
Please import `firebase` and add the types for this, do not use `any` if it's not necessary.
```suggestion
            db.ref(this.userTodosPath).on("value", (snap: firebase.database.DataSnapshot) => {
```

Reply:
^ check this mi loco